CRM/Core/BAO/Setting.php

```diff
@@ -199,6 +199,10 @@ public static function setItems(&$params, $domains = NULL) {
     $fieldsToSet = self::validateSettingsInput($params, $fields);
 
     foreach ($fieldsToSet as $settingField => &$settingValue) {
+      if (empty($fields['values'][$settingField])) {
+        Civi::log()->warning('Deprcated Path There is a setting (' . $settingField . ') not correctly defined. You may see unpredictabilitiy due to this, CRM_Core_Setting::setItems', array('civi.tag' => 'deprecated'));
```

Comment:
Sounds fair, but I think the key is to define _when_ it will be removed, it's very simple then for anyone who sees it after that version is released to delete. If we're not specific we can upset developer who could claim they didn't get enough notice.